prototypes/docking_wf_atomised.py

```diff
@@ -0,0 +1,82 @@
+from dask import delayed
+
+# inputs to the delayed functions are
+# probably length 1 lists of the objects
+
+@delayed
+def prep_targets(target: List[Target]):
+    pp: ProteinPrepper = ProteinPrepper(target)
+    prepped_targets: List[Target] = pp.prep_all()
+
+    return prepped_targets
+
+@delayed
+def prep_ligands(ligand: List[Ligand]):
+    lp: LigandPrepper = LigandPrepper(ligand)
+    prepped_ligands: List[Ligand] = lp.prep_all()
+    return prepped_ligands
+
+
+@delayed
+def docking_wf(docking_inputs: List[DockingInput]):
+
+   docker: Docker_OE = Docker_OE(docking_inputs)
+   docking_results: List[DockingResults] = docker.dock_all()
+
+   return docking_results
+
+
+
+# main
+
+def main():
+    ligand_fr = FileReader(ligand_fn, type="ligand", format='sdf')
+    target_fr = FileReader(target_fn, type="target", format='pdb')
+
+    prepped_targets = []
+    for target in target_fr.load():
+        prep_t = prep_target(ligand: Ligand)
+        prepped_targets.append(prep_t)
+
+    prepped_ligands = []
+    for ligand in ligand_fr.load():
+        prep_l = prep_ligand(ligand: Ligand)
+        prepped_ligands.append(prep_l)
+
+    # barrier: could make parallel here as well possibly as do not rely on collected results
+    ligand_selector: LigandSelector = PairwiseLigandSelector(prepped_ligands, prepped_targets)
```

Comment:
You could do a:

```suggestion
    ligand_selector: LigandSelector = delayed(PairwiseLigandSelector)(prepped_ligands, prepped_targets)
```
 and then `ligand_selector` is also a `Delayed` object. Any subsequent calls on that object will also give back `Delayed` objects.